lib/engine/matchers/all_keys_matcher.rb

```diff
@@ -15,7 +15,7 @@ def initialize
     # @param key [string] key value to be matched
     #
     # @return [boolean] true for all instances
-    def match?(_key, _attributes)
+    def match?(_key, _split_treatment, _attributes)
```

Comment:
I see no reason why pass client here, client stores impressions/metrics and return evaluations, whereas treatment just return evaluations and we need only this